extensions-core/protobuf-extensions/src/main/java/org/apache/druid/data/input/protobuf/FileBasedProtobufBytesDecoder.java

```diff
@@ -98,35 +98,40 @@ public DynamicMessage parse(ByteBuffer bytes)
         url = new URL(descriptorFilePath);
       }
       catch (MalformedURLException e) {
-        throw new ParseException(e, "Descriptor not found in class path or malformed URL:" + descriptorFilePath);
+        throw new ParseException(
+            descriptorFilePath,
+            e,
+            "Descriptor not found in class path or malformed URL:" + descriptorFilePath
+        );
       }
       try {
         fin = url.openConnection().getInputStream();
       }
       catch (IOException e) {
-        throw new ParseException(e, "Cannot read descriptor file: " + url);
+        throw new ParseException(url.toString(), e, "Cannot read descriptor file: " + url);
       }
     }
     DynamicSchema dynamicSchema;
     try {
       dynamicSchema = DynamicSchema.parseFrom(fin);
     }
     catch (Descriptors.DescriptorValidationException e) {
-      throw new ParseException(e, "Invalid descriptor file: " + descriptorFilePath);
+      throw new ParseException(descriptorFilePath, e, "Invalid descriptor file: " + descriptorFilePath);
     }
     catch (IOException e) {
-      throw new ParseException(e, "Cannot read descriptor file: " + descriptorFilePath);
+      throw new ParseException(descriptorFilePath, e, "Cannot read descriptor file: " + descriptorFilePath);
     }
 
     Set<String> messageTypes = dynamicSchema.getMessageTypes();
     if (messageTypes.size() == 0) {
-      throw new ParseException("No message types found in the descriptor: " + descriptorFilePath);
+      throw new ParseException(descriptorFilePath, "No message types found in the descriptor: " + descriptorFilePath);
     }
 
     String messageType = protoMessageType == null ? (String) messageTypes.toArray()[0] : protoMessageType;
     Descriptors.Descriptor desc = dynamicSchema.getMessageDescriptor(messageType);
     if (desc == null) {
       throw new ParseException(
+          null,
```

Comment:
does it make sense to use messageType instead of null?

Reply:
Made this use messageType